src/main/java/ch/uzh/ifi/hase/soprafs21/objects/GameEngine.java

```diff
@@ -243,6 +243,17 @@ public void deleteGameByGameID(UUID GameID){
         }
     };
 
+    public UUID findGameIdByPlayerName(String playerName){
+        for(Game game:runningGamesList){
+            for(Player player: game.getPlayerList()){
+                if(player.getPlayerName().equals(playerName)){
+                    return game.getGameId();
+                }
+            }
+        }
+        return null;
+    }
```

Comment:
@edischmitz HashMap is always better for searching, why did you shift to linear search with for loops? What was the short coming with the Hashmap implementation?

Reply:
@iamsiddhantsahu The HashMap Implementation required the use of an extra data Structure. This implementation is more efficient from a storage point of view. The Linear search is also not much less efficient since we are dealing with a small user base.